### PR TITLE
SIMD feature for backends

### PR DESCRIPTION
**Pull request by leonardoalt**
Depends on https://github.com/powdr-labs/powdr/pull/1923

Now that we use nightly, we can more easily enable AVX, Neon etc in backends that support it.

Not sure why fmt added some formatting changes.

### CODE REVIEW

File: backend/Cargo.toml [hunk 1]
```diff
@@ -29,12 +39,16 @@ pil-stark-prover = { git = "https://github.com/powdr-labs/pil-stark-prover.git",
 
 # TODO change this once Halo2 releases 0.3.1
```
Comment (georgwiese) [inline backend/Cargo.toml:40]: Don't we have to add `features = ["avx512"]` to `starky`, to be able to compile with `target-cpu=native`?

File: backend/Cargo.toml [hunk 2]
```diff
@@ -9,10 +9,20 @@ repository = { workspace = true }
...
+  "dep:halo2_curves",
+  "dep:snark-verifier",
+  "dep:halo2_solidity_verifier",
+]
 estark-polygon = ["dep:pil-stark-prover"]
 plonky3 = ["dep:powdr-plonky3"]
 
+# Enable AVX or Neon accordingly in backends that support them.
+plonky3-simd = ["powdr-plonky3?/nightly-features"]
+estark-starky-simd = ["starky/avx512"]
```
Comment (leonardoalt) [inline backend/Cargo.toml:23]: Its here @georgwiese

File: backend/Cargo.toml (hunk 1 above)
Comment (georgwiese) [reply backend/Cargo.toml:40]: Hmm, I see, it is activated via the `simd` feature.

It's just a bit weird that when running with `RUSTFLAGS='-Ctarget-cpu=native'`, it fails to compile if you don't activate the `simd` feature. But I don't see a better solution...

Comment (leonardoalt) [reply backend/Cargo.toml:40]: It is activated via the simd feature, if you want for all backends, or just the starky one.

Yea also don't have a better alternative, it's the same for plonky3.

**Review by georgwiese — COMMENTED** [review]
I think the instructions [here](https://docs.powdr.org/installation.html) don't work anymore, we need to add the `simd` feature. Could you update the docs?

File: pipeline/Cargo.toml [hunk 3]
```diff
@@ -8,10 +8,11 @@ homepage = { workspace = true }
 repository = { workspace = true }
 
 [features]
-default = [] # halo2 is disabled by default
+default = []                                      # halo2 is disabled by default
```
[the remark this answers is not in the document]
Comment (leonardoalt) [reply pipeline/Cargo.toml:11]: somehow fmt does that, I don't know why

**Comment by leonardoalt** [comment]
updated now

File: book/src/installation.md [hunk 4]
```diff
@@ -19,10 +19,12 @@ installed with the "Desktop Development With C++" Workloads option.
 
 If you want to enable feature `estark-polygon`, you also need the following
 runtime dependencies installed on the system:
+
```
Comment (leonardoalt) [inline book/src/installation.md:22]: fmt did this and the one below

**Review by georgwiese — DISMISSED** [review]
LGTM

File: plonky3/Cargo.toml [hunk 5]
```diff
@@ -13,6 +13,15 @@ name = "powdr_plonky3"
 name = "gen_poseidon_bb_consts"
 path = "src/bin/gen_poseidon_bb_consts.rs" 
 
+[features]
+default = []
+nightly-features = [
+  "p3-goldilocks/nightly-features",
+  "p3-baby-bear/nightly-features",
```
Comment (georgwiese) [inline plonky3/Cargo.toml:20]:
```suggestion
  "p3-baby-bear/nightly-features",
  "p3-koala-bear/nightly-features",
```

**Review by georgwiese — COMMENTED** [review]
Sorry for the back and forth with this PR, but now it makes sense to add KoalaBear ^^

File: plonky3/Cargo.toml (hunk 5 above)
Comment (leonardoalt) [reply plonky3/Cargo.toml:20]: done

**Review by georgwiese — APPROVED** [review]

**Pull request merged**